sky/backends/onprem_utils.py

```diff
@@ -492,6 +494,10 @@ def check_local_cloud_args(cloud: Optional[str] = None,
                 '`cloud: local` or no cloud in YAML or CLI args.')
         return True
     else:
+        if (len(enabled_clouds) == 1 and
+                isinstance(enabled_clouds[0], clouds.Local)):
+            raise click.UsageError('Cloud access is not set up. '
+                                   'Run: `sky check`')
```

Comment:
This reminds me that in the optimizer, we use `len(enabled_clouds) == 0` to test whether there is a cloud enabled. Does the on-prem make the `enabled_clouds` have 1 element by default? Should we update the function in the optimizer?
https://github.com/skypilot-org/skypilot/blob/e8c0772ee0e5dbe059cf7e73bc4441071026994f/sky/optimizer.py#L829-L832

Also, can we try to run the `check.check(quiet=True)` ourselves as the code shown above to free the user for running `sky check` manually if they already have one cloud setup.
```
def _is_cloud_disabled():
  return (len(enabled_clouds) > 1 and
                isinstance(enabled_clouds[0], clouds.Local)):
if _is_cloud_disabled():
  check.check(quiet=True)
  if _is_cloud_disabled():
    raise click.UsageError('Cloud access is not set up. '
                                   'Run: `sky check`')
```

Reply:
Done, thanks!